src/web_app/js/peerconnectionclient.js

```diff
@@ -38,6 +38,7 @@ var PeerConnectionClient = function(params, startTime) {
   this.pc_.onsignalingstatechange = this.onSignalingStateChanged_.bind(this);
   this.pc_.oniceconnectionstatechange =
       this.onIceConnectionStateChanged_.bind(this);
+  this.pc_.onnegotiationneeded = this.setupCallStats_();
```

Comment:
It's the earliest it can be attached to the peerConnection according the callstats api documentation. Since we did not use that event for anything I thought we might as well use it.